src/catalog/src/system_schema/information_schema/flows.rs

```diff
@@ -0,0 +1,306 @@
+            option_groups: StringVectorBuilder::with_capacity(INIT_CAPACITY),
+        }
+    }
+
+    /// Construct the `information_schema.views` virtual table
+    async fn make_views(&mut self, request: Option<ScanRequest>) -> Result<RecordBatch> {
+        let catalog_name = self.catalog_name.clone();
+        let predicates = Predicates::from_scan_request(&request);
+
+        let flow_info_manager = self.flow_metadata_manager.clone();
+
+        // TODO(discord9): use `AsyncIterator` once it's stable-ish
+        let mut stream = flow_info_manager
+            .flow_name_manager()
+            .flow_names(&catalog_name)
+            .await;
+
+        while let Some((flow_name, flow_id)) = stream
+            .try_next()
+            .await
+            .map_err(BoxedError::new)
+            .context(ListFlowsSnafu {
+                catalog: catalog_name.to_string(),
+            })?
+        {
+            let flow_info = flow_info_manager
+                .flow_info_manager()
+                .get(flow_id.flow_id())
+                .await
+                .map_err(BoxedError::new)
+                .context(InternalSnafu)?
+                .context(FlowInfoNotFoundSnafu {
+                    catalog_name: catalog_name.to_string(),
+                    flow_name: flow_name.to_string(),
+                })?;
+            self.add_view(&predicates, flow_id.flow_id(), flow_info)?;
+        }
+
+        self.finish()
+    }
+
+    fn add_view(
+        &mut self,
+        predicates: &Predicates,
+        flow_id: FlowId,
+        flow_info: FlowInfoValue,
+    ) -> Result<()> {
+        let row = [
+            (FLOW_NAME, &Value::from(flow_info.flow_name().to_string())),
+            (FLOW_ID, &Value::from(flow_id)),
+            (
+                CATALOG_NAME,
+                &Value::from(flow_info.catalog_name().to_string()),
+            ),
+        ];
+        if !predicates.eval(&row) {
+            return Ok(());
+        }
+        self.flow_names.push(Some(flow_info.flow_name()));
+        self.flow_ids.push(Some(flow_id));
+        self.catalog_names.push(Some(flow_info.catalog_name()));
+        self.raw_sqls.push(Some(flow_info.raw_sql()));
+        self.comments.push(Some(flow_info.comment()));
+        self.expire_afters.push(flow_info.expire_after());
+        self.source_table_id_groups.push(Some(
+            &serde_json::to_string(flow_info.source_table_ids()).context(JsonSnafu {
+                input: format!("{:?}", flow_info.source_table_ids()),
+            })?,
+        ));
+        self.sink_table_names
+            .push(Some(&flow_info.sink_table_name().to_string()));
+        self.flownode_id_groups.push(Some(
+            &serde_json::to_string(flow_info.flownode_ids()).context({
+                JsonSnafu {
+                    input: format!("{:?}", flow_info.flownode_ids()),
+                }
+            })?,
+        ));
+        self.option_groups
```

Comment:
I think joining the `options` just like `k1=v1, k2=v2` is enough.

Reply:
> I think joining the `options` just like `k1=v1, k2=v2` is enough.

i thought making it using proper json format might help with compatibility once we support json type, but since its a simple enough serde impl manually also make sense i guess